src/reporting/market.se

```diff
@@ -88,14 +88,27 @@ def initialize(reportingWindow: address, endTime: int256, numOutcomes: int256, p
     i = 0
     while(i < numOutcomes):
         self.shareTokens[i] = SHARE_TOKEN_FACTORY.createShareToken(self.controller)
+        self.shareTokens[i].initialize(self, i)
         i += 1
     self.approveSpenders()
+    require(self.getBranch().isContainerForShareToken(denominationToken) or self.controller.lookup('cash') == denominationToken)
```

Comment:
I wonder if we should reverse the order here as an optimization?  In _most_ cases, the token will be `cash` and by having the comparison against cash be first we'll avoid the share token checks in those cases.